openedx/core/djangoapps/theming/views.py

```diff
@@ -0,0 +1,104 @@
+"""
+Views file for the Darklang Django App
+"""
+
+from django.template.loader import render_to_string
+from django.utils.translation import ugettext as _
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from openedx.core.djangoapps.user_api.preferences.api import (
+    delete_user_preference,
+    get_user_preference,
+    set_user_preference,
+)
+from openedx.core.djangoapps.util.user_messages import (
+    register_error_message,
+    register_success_message,
+)
+from web_fragments.fragment import Fragment
+
+from .helpers import theme_exists
+from .models import SiteTheme
+
+PREVIEW_SITE_THEME_PREFERENCE_KEY = 'preview-site-theme'
+
+
+def get_user_preview_site_theme(request):
+    """
+    Returns the preview site for the current user, or None if not set.
+    """
+    user = request.user
+    if not user or user.is_anonymous():
+        return None
+    preview_site_name = get_user_preference(user, PREVIEW_SITE_THEME_PREFERENCE_KEY)
+    if not preview_site_name:
+        return None
+    return SiteTheme(site=request.site, theme_dir_name=preview_site_name)
+
+
+def set_user_preview_site_theme(request, preview_site_theme):
+    """
+    Sets the current user's preferred preview site theme.
+
+    Args:
+        request: the current request
+        preview_site_theme: the preview site theme (or None to remove it)
```

Comment:
Hmmm.  I don't see a response and I don't see this code changed.  Is it possible some changes got lost, or did all my comments get lost in the shuffle?